### PR TITLE
Search

### PR DESCRIPTION
**Pull request by marmarta**
This is a search implementation based on initial @ninavizz design and further development by me. It is based on #15 because I want to avoid potential rebase hell.

### CODE REVIEW

File: qubes_menu/page_handler.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+#
+# You should have received a copy of the GNU Lesser General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+"""Abstract Menu page"""
+import abc
+
+
+class MenuPage(abc.ABC):
+    """Abstract Menu Page."""
+    def initialize_page(self):
```
[the remark this answers is not in the document]
Comment (marmarta) [reply qubes_menu/page_handler.py:27]: done

**Comment by marmarta** [comment]
PipelineRetryFailed

**Comment by marmarta** [comment]
Added screenshots to the readme, can see them here: https://github.com/marmarta/qubes-desktop-linux-menu/tree/search_alpha#search-pane

**Review by marmarek — CHANGES_REQUESTED** [review]

**Comment by marmarek** [comment]
Regarding keyboard navigation:
 - when switching VM selection, an application from previous VM remain selected - this behaves funny regarding subsequent navigation on the apps pane (and also, pressing enter does launch what was selected before but now invisible)
 - it isn't clear if focus is on apps or VMs - fixing this by unselecting app when going to VMs pane may fix the previous point too
 - arrow up cannot switch back to the search bar - not sure if an issue, since typing anything does switch
 -

**Comment by marmarta** [comment]
Fixed filter to be more resilient + fixed keyboard nav; left the "can't explicitly move focus to search bar" issue, if it turns out to be problematic in use, will return to change it.

**Review by marmarek — APPROVED** [review]

**Pull request merged**